growth/too/tasks/growthdb_cgi.py

```diff
@@ -19,14 +21,105 @@
 """
 
 
+def prepare_candidates_for_object_table(sources_all):
+    """
+    Prepare a list of disctionaries with relevant info
+    for each source in the marshal
+
+    Parameters
+    ----------
+    sources_all : list(sqlalchemy.util._collections.result)
+        List of sources queried from the local database.
+
+    Returns
+    -------
+    sources_growth_marshal: list(dict)
+        list of dictionaries with values formatted to be
+        correctly displayed in the objects page.
+
+    Notes
+    -----
+    The main action of this function is the creation of strings
+    including the source coordinates in sexagesimal format.
+    """
+
+    sources_growth_marshal = []
+    for s, l in sources_all:
+        s_dict = s.__dict__
+        s_dict.update(l.__dict__)
+        s_dict["first_detection_time"] = l.first_detection_time
+        s_dict["first_detection_mag"] = l.first_detection_mag
+        s_dict["first_detection_magerr"] = l.first_detection_magerr
+        s_dict["first_detection_instrument"] = l.first_detection_instrument
+        s_dict["first_detection_filter"] = l.first_detection_filter
+        s_dict["latest_detection_time"] = l.latest_detection_time
+        s_dict["latest_detection_mag"] = l.latest_detection_mag
+        s_dict["latest_detection_magerr"] = l.latest_detection_magerr
+        s_dict["latest_detection_instrument"] = l.latest_detection_instrument
+        s_dict["latest_detection_filter"] = l.latest_detection_filter
+
+        if ('ra' in s_dict) and ('dec' in s_dict):
+            try:
+                rr, dd = s_dict['ra'].deg, s_dict['dec'].deg
+            except (TypeError, KeyError, ValueError, AttributeError):
+                rr, dd = s_dict['ra'], s_dict['dec']
+            s_coords = SkyCoord(ra=rr*u.deg, dec=dd*u.deg)
+            s_dict["ra"], s_dict["dec"] = s_coords.ra, s_coords.dec
+            ra_h = '{0:02d}'.format(int(s_coords.ra.hms.h))
+            ra_m = '{0:02d}'.format(int(s_coords.ra.hms.m))
+            ra_s = '{0:02d}'.format(int(s_coords.ra.hms.s))
+            ra_sd = '{0:02d}'.format(int(100*(s_coords.ra.hms.s -
+                                     int(s_coords.ra.hms.s))))
+            s_dict["ra_string"] = f"{ra_h}:{ra_m}:{ra_s}.{ra_sd}"
+            dec_d = '{0:02d}'.format(int(s_coords.dec.dms.d))
+            dec_m = '{0:02d}'.format(abs(int(s_coords.dec.dms.m)))
+            dec_s = '{0:02d}'.format(int(abs(s_coords.dec.dms.s)))
+            dec_sd = '{0:02d}'.format(int(100 * abs(s_coords.dec.dms.s -
+                                      int(s_coords.dec.dms.s))))
+            s_dict["dec_string"] = f"{dec_d}:{dec_m}:{dec_s}.{dec_sd}"
```

Comment:
Please use `SkyCoord`'s string formatting methods to build the sexagesimal string representation rather than doing it yourself.

Reply:
Done.